src/main/java/ratpack/zipkin/SpanNameProvider.java

```diff
@@ -0,0 +1,15 @@
+package ratpack.zipkin;
+
+/**
+ * Functional interface for span name customizers.
+ */
+@FunctionalInterface
+public interface SpanNameProvider {
+  /**
+   * Given a request and response, return the name to be used for the Span.
+   * @param request the request
+   * @param response the response
+   * @return the Span name
+   */
+  String spanName(ServerRequest request, ServerResponse response);
```

Comment:
response is odd to have here. seems the reason is to expose the path binding.. if so, make a comment or doc it so folks don't forget

Reply:
@adriancole hmmm...yeah I see your point. TBH, there's probably not a hell of a lot of value in having `ServerResponse` here (I think I just have it this way because it seemed more "symmetrical"). I could just put the `PathBinding` in there - at least then it's pretty clear that that's the *only* reason that parameter exists (i.e. don't expect to be able to get anything else from the response).